qiskit_experiments/data_processing/data_processor.py

```diff
@@ -125,79 +190,117 @@ def _call_internal(
                 then all nodes in the data processing chain will be called.
 
         Returns:
-            datum_ and history if with_history is True or datum_ if with_history is False.
+            When ``with_history`` is ``False`` it returns an numpy array of processed data.
+            Otherwise it returns a tuple of above with a list of intermediate data at each step.
         """
         if call_up_to_node is None:
             call_up_to_node = len(self._nodes)
 
-        datum_, error_ = self._data_extraction(data), None
+        data = self._data_extraction(data)
 
         history = []
-        for index, node in enumerate(self._nodes):
-
-            if index < call_up_to_node:
-                datum_, error_ = node(datum_, error_)
-
-                if with_history and (
-                    history_nodes is None or (history_nodes and index in history_nodes)
-                ):
-                    history.append((node.__class__.__name__, datum_, error_, index))
+        for index, node in enumerate(self._nodes[:call_up_to_node]):
+            data = node(data)
+
+            if with_history and (history_nodes is None or index in history_nodes):
+                history.append(
+                    (
+                        node.__class__.__name__,
+                        unp.nominal_values(data),
+                        unp.std_devs(data),
```

Comment:
Won't this need some form of type checking?

Reply:
good catch. done in https://github.com/Qiskit/qiskit-experiments/pull/481/commits/a1389e92366b65b06c34bb1a47aa9fd0d2621b30